lib/Importer/ProtobufLoader.cpp

```diff
@@ -15,18 +15,53 @@
  */
 
 #include "glow/Importer/ProtobufLoader.h"
-
+#include "llvm/Support/CommandLine.h"
 #include <string>
 
 namespace glow {
 
+llvm::cl::OptionCategory loaderOptCat("Model Loader Options");
+
+static llvm::cl::opt<bool> isConstFoldLoaderOps(
+    "const-fold-ops",
+    llvm::cl::desc(
+        "Performs constant folding on ONNX and Caffe Operators while loading."),
+    llvm::cl::init(false), llvm::cl::cat(loaderOptCat));
+
 bool isArrayConstant(llvm::ArrayRef<size_t> a) {
   for (size_t i = 1; i < a.size(); i++)
     if (a[0] != a[i])
       return false;
   return true;
 }
 
+void setConstantFoldLoaderOpsFlag(bool flag) { isConstFoldLoaderOps = flag; }
+
+bool getConstantFoldLoaderOpsFlag() { return isConstFoldLoaderOps; }
+
+bool ProtobufLoader::isConstantFoldable(llvm::SmallVector<NodeValue, 4> &inputs,
+                                        std::string typeName) const {
+  int numInputs = inputs.size();
+  if (!getConstantFoldLoaderOpsFlag()) {
+    return false;
+  }
+  // foldUnsupportedTypes: List of typenames unsupported for folding.
+  std::string foldUnsupportedTypes[] = {"Constant"};
+  std::string *foo = std::find(std::begin(foldUnsupportedTypes),
+                               std::end(foldUnsupportedTypes), typeName);
+  // Early exit if folding is not supported for current operator.
+  if (foo == std::end(foldUnsupportedTypes)) {
```

Comment:
May be reverse the condition and perform an early return. This avoid the big indented block which is currently the body of the `if`-statement.